### PR TITLE
Refactor autotool files

### PR DESCRIPTION
**Pull request by xgarrido**
This PR removes the auto-generated files from the repository and makes sure the `configure`, `Makefile` are properly generated by the host computer.

### CODE REVIEW

**Comment by xgarrido** [comment]
To ease the developer work, we can imagine to add an `autogen.sh` file (as it is usually done for C program) whose holds the following commands 

```
autoreconf -ivf
./configure
make clean
make
```

File: .gitignore [hunk 1]
```diff
@@ -1,11 +1,23 @@
 # Ignore autoconf build files.
 Makefile
+Makefile.in
 config.h
 config.log
 config.status
 libtool
 stamp-h1
 autom4te.cache
```
Comment (damonge) [inline .gitignore:9]: Why not remove the files above? Aren't those generated by autoreconf?

**Review by damonge — APPROVED** [review]
LGTM!

**Comment by damonge** [comment]
Thanks a lot @xgarrido

**Pull request merged**